bids/layout/utils.py

```diff
@@ -0,0 +1,78 @@
+import json
+from pathlib import Path
+from packaging.version import Version
+
+
+# As per https://bids.neuroimaging.io/bids_spec1.1.1.pdf
+desc_fields = {
+    Version("1.1.1"): {
+        "required": ["Name", "BIDSVersion"],
+        "recommended": ["License"],
+        "optional": ["Authors", "Acknowledgements", "HowToAcknowledge",
+                     "Funding", "ReferencesAndLinks", "DatasetDOI"]
+    }
+}
+
+
+def get_description_fields(version, type_):
+    if isinstance(version, str):
+        version = Version(version)
+    if not isinstance(version, Version):
+        raise TypeError("Version must be a string or a packaging.version.Version object.")
+
+    if version in desc_fields:
+        return desc_fields[version][type_]
+    return desc_fields[max(desc_fields.keys())][type_]
```

Comment:
This seems brittle. What about 1.0.0, 1.0.1, ... 1.2.1?

Reply:
Well, I tested that the "<" operator was well defined for Version() objects so that the max function works as expected. It seems to do good job:

In [76]: from packaging.version import 
In [77]: Version("1.0.0") < Version("1.0.1")                                                                                                                            
Out[77]: True
In [78]: Version("1.0.0") < Version("1.2.1")                                                                                                                            
Out[78]: True
In [79]: Version("0.9.0") < Version("1.0.0")                                                                                                                            
Out[79]: True
In [80]: Version("0.9.0") < Version("1.0.0b")                                                                                                                           
Out[80]: True
In [81]: Version("0.9.0") < Version("1.0.0b")                                                                                                                           
Out[81]: True
In [82]: Version("1.9.0") < Version("1.0.0b")                                                                                                                           
Out[82]: False

Now, if you start using very weird version numbering standard like "1.b.0b", then it throws an exception InvalidVersion, which seems fair to me. Anyway, if you want to propose a less brittle alternative, your suggestions will be welcome.

Reply:
Oh, sorry, I didn't really read the `max()` bit properly. I need to look more closely.